lib/metadata_json_deps/forge_helper.rb

```diff
@@ -32,6 +32,30 @@ def get_metadata_json(name)
       metadata
     end
 
+    def get_metadata_json_uri(module_name)
+      uri = URI.parse('https://forgeapi.puppetlabs.com/v3/modules/' + module_name)
+      request = Net::HTTP::Get.new(uri.to_s)
+      request.content_type = 'application/json'
+      req_options = {
+          use_ssl: uri.scheme == 'https',
+      }
+
+      response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
+        http.request(request)
+      end
+
+      response.body
+    end
+
+    def check_deprecated_at(module_name)
```

Comment:
This would be better incorporated as part of the `check_deprecated` method on line 67 `runner.rb` . On line 22 `runner.rb` you can then pass in the module name and remove the need for the `@forge.check_deprecated_at(...)` check.